src/maybe.js

```diff
@@ -14,35 +14,23 @@ class Maybe {
   }
 
   map(fn) {
-    if (this.isNothing()) {
-      return Maybe.of(null);
-    }
-    return Maybe.of(fn(this.value));
+    return Maybe.of(this.isNothing() ? null : fn(this.value));
   }
 
   mapAll(fnList) {
-    const maybeContext = this;
-    if (fnList === null || fnList === undefined) {
-      return maybeContext;
-    } else if (fnList.constructor === Array) {
-      return fnList.reduce((prevValue, currVal) => {
-        if (isFunction(currVal)) {
-          return prevValue.map(currVal);
-        }
-        return prevValue;
-      }, maybeContext);
+    if (isArray(fnList)) {
+      return fnList.reduce(
+        (prevValue, currVal) => (isFunction(currVal) ? prevValue.map(currVal) : prevValue),
+        this,
```

Comment:
I may have skimped a bit on the tests. I think it would be worth writing (or modifying) an existing test to illustrate that the correct context (`this`) is being applied.

Reply:
Something like:

```
    it('executes an array of functions', () => {
      const testFns = [
        val => Object.assign({}, val, { monadType: 'maybe' }),
        val => val,
        val => val,
      ];
      const actual = Maybe.of({ username: 'strange-developer' }).mapAll(
        testFns
      );
      expect(actual).to.deep.eq(
        Maybe.of({ username: 'strange-developer', monadType: 'maybe' })
      );
    });
```

Reply:
Sweet, will add it tonight when I get to coding again.

Reply:
Cool, added.